feat(STOCK-52)

Blaskowitz14 opened the pull request:
- Added sync with tg-bot information and updates

Blaskowitz14 committed:
feat(STOCK-52)
- Added sync with tg-bot information and updates

vladdossik commented on src/test/java/lissa/trading/user/service/service/UserCreationServiceImplTest.java:
```diff
@@ -0,0 +1,193 @@
+
+    @Test
+    void createUserFromTempUserReg_success() {
+        User user = new User();
```
все создания нужно вынести в отдельный класс и не засорять тесты

Blaskowitz14 replied:
Я кстати тесты не писал
Хз почему они от моего имени запушились
Поправлю

vladdossik commented on src/main/java/lissa/trading/user/service/config/RabbitConfig.java:
```diff
@@ -15,18 +19,63 @@ public class RabbitConfig {
+    public Binding favoriteStocksBinding(Queue userFavoriteStocksQueue, TopicExchange userExchange) {
+        return BindingBuilder.bind(userFavoriteStocksQueue)
+                .to(userExchange)
+                .with("*.favourite-stocks");
```
в йамл

vladdossik commented on src/main/java/lissa/trading/user/service/config/RabbitConfig.java:
```diff
@@ -15,18 +19,63 @@ public class RabbitConfig {
+    public Binding userUpdateBinding(Queue userUpdateQueue, TopicExchange userExchange) {
+        return BindingBuilder.bind(userUpdateQueue)
+                .to(userExchange)
+                .with("*.user-update");
```
в йамл

vladdossik commented on src/main/java/lissa/trading/user/service/dto/notification/UserFavoriteStocksUpdateDto.java:
```diff
@@ -0,0 +1,19 @@
+@Builder
+public class UserFavoriteStocksUpdateDto {
+    private UUID externalId;
+    List<FavoriteStocksEntity> favoriteStocksEntity;
```
private

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -54,7 +69,11 @@ public void blockUserByTelegramNickname(String telegramNickname) {
     @Transactional
     public void deleteUserByExternalId(UUID externalId) {
-        userRepository.delete(findUserByExternalId(externalId));
+        User deletedUser = findUserByExternalId(externalId);
```
теперь вместо одного запроса к бд тут 2
нужно ли было так менять хз хз хз

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -93,6 +112,38 @@ public List<UUID> getUserIdsWithPaginationAndFilters(Pageable pageable, String f
+
+    private void updateUserBroker(User user) {
+        String tinkoffToken = user.getTinkoffToken();
+        if (tinkoffToken.startsWith("t.")
```
это наверное должно быть не тут + не должно быть магических констант

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/TinkoffUpdateServiceImpl.java:
```diff
@@ -34,8 +45,29 @@ public class TinkoffUpdateServiceImpl implements UpdateService {
+        return BROKER;
+    }
+
+    public void fullUserUpdate(User user) {
```
название метода маленька говно

vladdossik commented on src/main/resources/application-local.yaml:
```diff
@@ -42,6 +42,16 @@ integration:
     statistics-service:
-      user-queue: user-stats-queue
\ No newline at end of file
+      user-queue: user-stats-queue
```
мб тогда уж и для statistics-service поправить конфиг чтобы было как для user-service

Blaskowitz14 committed:
feat(STOCK-52)
- massive bugfixes
- accidentally omitted code
- naming & structure & logic changes
- tests rework

Blaskowitz14 committed:
feat(STOCK-52)
- forgot to change value in stats publisher

vladdossik commented on src/main/java/lissa/trading/user/service/config/RabbitConfig.java:
```diff
@@ -12,21 +16,72 @@
+    }
+
+    @Bean
+    public Queue userFavoriteStocksQueue() {
```
старайся указывать сначала все Queue потом все TopicExchange и тд

или же разделяй по конкретной задаче
тут как будто вообще нет логики

vladdossik commented on src/main/java/lissa/trading/user/service/controller/InternalController.java:
```diff
@@ -152,4 +154,26 @@ public void blockUserByTelegramNickname(@PathVariable String telegramNickname) {
+            responseCode = "200",
+            content = @Content()
+    )
+    @PostMapping("/update/favouriteStocks/{externalId}")
```
может возможно какую-то часть урлы вынести в маппинг наверх

vladdossik commented on src/main/java/lissa/trading/user/service/utils/Tokens.java:
```diff
@@ -0,0 +1,28 @@
+import lissa.trading.lissa.auth.lib.security.EncryptionService;
+import lissa.trading.user.service.service.update.factory.SupportedBrokersEnum;
+
+public class Tokens {
```
название класса мне ни о чем не говорит

vladdossik commented on src/main/resources/application-local.yaml:
```diff
@@ -42,6 +42,19 @@ integration:
+      favourite-stocks-queue:
+        name: favourite-stocks
+        routing-key: userService.favourite-stocks
+        template: "*.favourite-stocks"
```
зачем тут template

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/factory/UpdateServiceFactory.java:
```diff
@@ -0,0 +1,26 @@
+@Component
+public class UpdateServiceFactory {
+
+    private final Map<SupportedBrokersEnum, UpdateService> UpdateServiceMap;
```
за название с большой буквы можно смело бить по еблу

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserService.java:
```diff
@@ -22,4 +25,10 @@ public interface UserService {
 
     List<UUID> getUserIdsWithPaginationAndFilters(Pageable pageable, String firstName,
                                                               String lastName);
+
```
получается лишний пробел

Blaskowitz14 replied:
![image](https://github.com/user-attachments/assets/42d6fad3-8c8d-45b2-901a-c3497cf1c061)

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -32,14 +45,23 @@ public class UserServiceImpl implements UserService {
+        User user = userMapper.updateUserFromDto(userUpdates, findUserByExternalId(externalId));
+        updateUserBroker(user);
+        userRepository.save(user);
+        log.info("updated user {}", user);
```
зачем выводить фулл юзера

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -54,7 +76,12 @@ public void blockUserByTelegramNickname(String telegramNickname) {
+        log.info("deleting user {}", externalId);
         userRepository.delete(findUserByExternalId(externalId));
+        User deletedUser = new User();
+        deletedUser.setExternalId(externalId);
```
это вообще че такое

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -93,6 +120,76 @@ public List<UUID> getUserIdsWithPaginationAndFilters(Pageable pageable, String f
+                    })
+                    .toList();
+
+            user.getFavoriteStocks().clear();
```
это можно вынести в сущность пользователя и назвать как-то типа setFavoriteStocks или типа того

vladdossik replied:
вместе со следующей строчкой

vladdossik commented on src/main/java/lissa/trading/user/service/service/UserServiceImpl.java:
```diff
@@ -93,6 +120,76 @@ public List<UUID> getUserIdsWithPaginationAndFilters(Pageable pageable, String f
+
+    private void updateUserBroker(User user) {
+        SupportedBrokersEnum broker = Tokens.determineTokenKind(user.getTinkoffToken());
+        {
```
нахуя тут фигурная скобка

vladdossik commented on src/main/java/lissa/trading/user/service/service/consumer/UserUpdatesNotificationConsumerImpl.java:
```diff
@@ -0,0 +1,102 @@
+    @Override
+    public void receiveUserUpdateNotification(UserUpdateNotificationDto userUpdateDto, @Header("amqp_receivedRoutingKey") String routingKey) {
+        log.info("received user update notification: {}", userUpdateDto);
+        if (routingKey.equals(userUpdateRoutingKey)) {
```
оставил коммент в тг боте по этому поводу

vladdossik commented on src/main/java/lissa/trading/user/service/service/publisher/NotificationContext.java:
```diff
@@ -0,0 +1,25 @@
+@RequiredArgsConstructor
+@Slf4j
+public class NotificationContext {
+    private final static ThreadLocal<Boolean> externalSourceContext = ThreadLocal.withInitial(() -> false);
```
зачем это нужно

vladdossik commented on src/main/java/lissa/trading/user/service/service/publisher/UserUpdatesPublisherImpl.java:
```diff
@@ -0,0 +1,61 @@
+    private final FavoriteStockMapper favoriteStockMapper;
+
+    @Value("${integration.rabbit.user-service.exchange.name}")
+    private String exchangeName;
```
сначала идут переменные, потом бины

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/MoexUpdateServiceImpl.java:
```diff
@@ -0,0 +1,111 @@
+    private final UserRepository userRepository;
+    private final UserUpdatesPublisher userUpdatesPublisher;
+
+    private final static SupportedBrokersEnum BROKER = SupportedBrokersEnum.MOEX;
```
наверх

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/MoexUpdateServiceImpl.java:
```diff
@@ -0,0 +1,111 @@
+
+    @Override
+    public void updateUserFavouriteStocks(User user, TickersDto tickersDto) {
+        if (tickersDto.getTickers() == null || tickersDto.getTickers().isEmpty()) {
```
CollectionUtils.isEmpty

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/MoexUpdateServiceImpl.java:
```diff
@@ -0,0 +1,111 @@
+
+        for (FavoriteStocksEntity favoriteStock : favoriteStocksEntities) {
+            if (favoriteStock.getServiceTicker().equals(getBroker().name())) {
+                Stock updatedStock = stockDtoMap.get(favoriteStock.getTicker());
```
может не объявлять переменную каждый раз ? 

vladdossik commented on src/main/java/lissa/trading/user/service/service/update/TinkoffUpdateServiceImpl.java:
```diff
@@ -34,8 +47,29 @@ public class TinkoffUpdateServiceImpl implements UpdateService {
+    private final UserUpdatesPublisher userUpdatesPublisher;
+    private final NotificationContext notificationContext;
+
+    private final static SupportedBrokersEnum BROKER = SupportedBrokersEnum.TINKOFF;
```
наверх

vladdossik commented on src/test/java/lissa/trading/user/service/service/BaseTest.java:
```diff
@@ -103,7 +124,9 @@ private User createUser() {
         userEntity.setLastName("Doe");
```
нужно вынести все создания пользователей и тд в какой-нибудь класс типа AbstractInitialization

Blaskowitz14 committed:
feat(STOCK-52)
- naming, structure and logic changes

Blaskowitz14 committed:
feat(STOCK-52)
- one letter changed

vladdossik commented on src/main/java/lissa/trading/user/service/controller/InternalController.java:
```diff
@@ -152,4 +144,26 @@ public void blockUserByTelegramNickname(@PathVariable String telegramNickname) {
+            responseCode = "200",
+            content = @Content()
+    )
+    @PostMapping("/update/favouriteStocks/{externalId}")
```
вообще слово update тут как бы нахуй не нужно

vladdossik commented on src/main/resources/application-local.yaml:
```diff
@@ -42,6 +42,28 @@ integration:
+          name: user-service-user-update
+          routing-key: userService.user-update
+    tg-bot:
+      queues:
```
зачем нам вообще вот этот вложенный queues:
и наверху тоже
и так ведь понятно че там внутри будет

+ сама структура кривая
спроси у Саши или Димы (до кого-то доебывался по поводу структуры
получается прям коряво и нужно переделать

Blaskowitz14 committed:
feat(STOCK-52)
- yaml refactor, minor naming changes

Blaskowitz14 committed:
feat(STOCK-52)
- Resolved conflicts

Blaskowitz14 committed:
feat(STOCK-52)
- minor changes
